### PR TITLE
Remove the bus from teamgroupsync

### PR DESCRIPTION
**Pull request by undef1nd**
**Special notes for your reviewer**:
Done as a part of the bus cleanup in Enterprise. Replaces the SyncTeamsCommand with a hook to stop using the bus.

### CODE REVIEW

File: pkg/services/teamgroupsync/teamgroupsync.go [hunk 1]
```diff
@@ -0,0 +1,24 @@
+package teamgroupsync
```
Comment (sakjur) [inline pkg/services/teamgroupsync/teamgroupsync.go:1]:
```suggestion
// Package teamgroupsync provides the interface for the Grafana Enterprise team sync feature.
package teamgroupsync
```

File: pkg/services/teamgroupsync/teamgroupsync.go [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+import (
+	"github.com/grafana/grafana/pkg/models"
+	"github.com/grafana/grafana/pkg/registry"
+)
+
+type TeamSync interface {
+	SyncTeams(user *models.User, externalUser *models.ExternalUserInfo) error
+}
+
+type TeamSyncService struct{}
```
Comment (sakjur) [inline pkg/services/teamgroupsync/teamgroupsync.go:12]:
```suggestion
type teamSyncStub struct{}
```

Personal preference, I don't think we need this to be public.

File: pkg/services/login/login.go [hunk 3]
```diff
@@ -109,10 +111,7 @@ func (ls *LoginService) UpsertUser(cmd *models.UpsertUserCommand) error {
 		}
 	}
 
-	err := ls.Bus.Dispatch(&models.SyncTeamsCommand{
-		User:         cmd.Result,
-		ExternalUser: extUser,
-	})
+	err := ls.TeamSync.SyncTeams(cmd.Result, extUser)
 	if err != nil && !errors.Is(err, bus.ErrHandlerNotFound) {
```
Comment (sakjur) [inline pkg/services/login/login.go:115]:
```suggestion
	if err != nil {
```

**Review by sakjur — COMMENTED** [review]
Leaving some early suggestions, nice work on getting the bus out :+1:

What is left before this can be reviewed?

File: pkg/services/teamgroupsync/teamgroupsync.go (hunk 1 above)
Comment (undef1nd) [reply pkg/services/teamgroupsync/teamgroupsync.go:1]: Great! Thank you. Totally forgot about package comments.

File: pkg/services/login/login.go (hunk 3 above)
Comment (undef1nd) [reply pkg/services/login/login.go:115]: Will do it once remove the other bus handlers from the service in Enterprise.

Comment (sakjur) [reply pkg/services/login/login.go:115]: Huh, is there any way that `err` can be `bus.ErrHandlerNotFound` here?

Comment (undef1nd) [reply pkg/services/login/login.go:115]: I thought yes, as long as the bus is still used inside of SyncTeams method in Enterprise. Do I get it wrong?

File: pkg/services/login/login_test.go [hunk 4]
```diff
@@ -74,6 +76,60 @@ func Test_syncOrgRoles_whenTryingToRemoveLastOrgLogsError(t *testing.T) {
...
+func Test_teamSync(t *testing.T) {
+	login := LoginService{
+		Bus:          bus.New(),
+		QuotaService: &quota.QuotaService{},
+	}
+
+	upserCmd := &models.UpsertUserCommand{ExternalUser: &models.ExternalUserInfo{Email: "test_user@gmail.com"}}
+	expectedUser := &models.User{
+		Id:    1,
+		Email: "test_user@gmail.com",
```
Comment (sakjur) [inline pkg/services/login/login_test.go:88]:
```suggestion
		Email: "test_user@example.org",
```

**Review by sakjur — APPROVED** [review]
LGTM! :+1:

**Pull request merged**